src/Common/Config/ConfigProcessor.cpp

```diff
@@ -792,6 +812,21 @@ void ConfigProcessor::decryptEncryptedElements(LoadedConfig & loaded_config)
 
 #endif
 
+XMLDocumentPtr ConfigProcessor::hideElements(LoadedConfig & loaded_config)
+{
+    XMLDocumentPtr new_preprocessed_xml = new Poco::XML::Document;
```

Comment:
`hideElements` is currently called with the preprocessed config but it must not be necessarily be so. We should call `new_preprocessed_xml` rather `xml_tree_copy`.

Reply:
Done